ext/nnstreamer/android_source/gstamcsrc.c

```diff
@@ -0,0 +1,976 @@
+    AMediaFormat *format = AMediaExtractor_getTrackFormat(priv->ex, i);
+    const char *mime;
+    if (AMediaFormat_getString(format, AMEDIAFORMAT_KEY_MIME, &mime)) {
+      if (!strncmp(mime, "video/", 6)) {
+        /** Use this video track */
+        LOGI ("Video track found: %s", AMediaFormat_toString(format));
+
+        priv->codec = AMediaCodec_createDecoderByType(mime);
+        priv->renderstart = -1;
+        priv->renderonce = TRUE;
+        priv->sawInputEOS = FALSE;
+        priv->sawOutputEOS = FALSE;
+        priv->isPlaying = FALSE;
+
+        AMediaFormat_getInt32(format, AMEDIAFORMAT_KEY_WIDTH, &priv->width);
+        AMediaFormat_getInt32(format, AMEDIAFORMAT_KEY_HEIGHT, &priv->height);
+        AMediaFormat_getInt32(format, AMEDIAFORMAT_KEY_FRAME_RATE, &priv->framerate);
+        AMediaFormat_getInt64(format, AMEDIAFORMAT_KEY_DURATION, &priv->duration);
+
+        AMediaCodec_configure(priv->codec, format, NULL /** surface */, NULL /** crypto */, 0);
+        AMediaExtractor_selectTrack(priv->ex, i);
+        AMediaFormat_delete(format);
+        
+        return TRUE;
+      }
+    } else 
+      LOGE ("No mime type"); 
+
+    AMediaFormat_delete(format);
+  }
+
+  return FALSE;
+}
+
+/**
+ * @brief open the target file and check its stats
+ */
+static gboolean gst_amc_src_media_open (GstAMCSrc *self)
+{
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+  struct stat stat_results;
+
+  if (priv->filename == NULL || priv->filename[0] == '\0')
+    goto error;
+  
+  LOGI ("opening file %s", priv->filename);
+  
+  priv->fd = open (priv->filename, O_RDONLY | O_BINARY, 0);
+  if (priv->fd < 0) {
+    LOGE ("Error: gst_open() failed");
+    goto error;
+  }
+
+  if (fstat (priv->fd, &stat_results) < 0) {
+    LOGE ("Error: fstat() failed");
+    goto error_close;
+  }
+
+  if (S_ISDIR (stat_results.st_mode) || S_ISSOCK (stat_results.st_mode)) {
+    LOGE ("Error: invalid file type");
+    goto error_close;
+  }
+
+  priv->pos = 0;
+  priv->size = stat_results.st_size;
+
+  if (S_ISREG (stat_results.st_mode))
+    priv->is_regular = TRUE;
+  
+  {
+    off_t res = lseek (priv->fd, 0, SEEK_END);
+
+    if (res < 0) {
+      LOGW ("disabling seeking, lseek failed: %s", g_strerror (errno));
+      priv->seekable = FALSE;
+    } else {
+      res = lseek (priv->fd, 0, SEEK_SET);
+      if (res < 0) {
+        priv->seekable = FALSE;
+        LOGE ("Error: lseek() failed");
+        goto error_close;
+      }
+      priv->seekable = TRUE;
+    }
+  }
+
+  priv->seekable = priv->seekable && priv->is_regular;
+
+  return gst_amc_src_codec_config (self);
+
+error_close:
+  close (priv->fd);
+error:
+  return FALSE;
+}
+
+/**
+ * @brief set file location for media codec
+ * @param[in] location Name of the media file
+ */
+static gboolean
+gst_amc_src_set_location (GstAMCSrc * self, const gchar * location,
+    GError ** err)
+{
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  if (location == NULL) {
```

Comment:
let's consider old file name (if se priv->filename is not null (old file name), need to free string.)

Reply:
and if old file was opened, it needs to close and clear the media info.

Reply:
You're right. It will be updated. FYI, a file descriptor is closed in advance when media codec is configured.